### DISCUSSION TITLE
[Feature Request] OIDC / OAuth2 option

### DISCUSSION THREAD

**Discussion by Aloe-recite** [ellite/Wallos#653] [Ideas]
Hi, as per title, I'd really love to be able to login using OIDC or OAuth2 instead of internal authentication. This would add security and enable me and my users to SSO (one less password to remember = wife is happy).
Thanks and keep up the good work!

**Comment by D3an1el** [comment]
Yeah i agree "happy wife, happy life". Also if this does get implemented, please have the option PKCE.

Reason for PKCE:

https://kanidm.github.io/kanidm/master/frequently_asked_questions.html#oauth2)

https://oauth.net/2/pkce/
